pubtools/_quay/iib_operations.py

```diff
@@ -0,0 +1,245 @@
+        "ssh_remote_host",
+        "ssh_user",
+        "ssh_password",
+        "docker_settings",
+    ]
+
+    for setting in required_settings:
+        if setting not in target_settings:
+            raise InvalidTargetSettings(
+                "'{0}' must be present in the target settings.".format(setting)
+            )
+
+    required_docker_settings = ["umb_urls", "docker_reference_registry"]
+    for setting in required_docker_settings:
+        if setting not in target_settings["docker_settings"]:
+            raise InvalidTargetSettings(
+                "'{0}' must be present in the docker settings.".format(setting)
+            )
+    if (
+        "iib_overwrite_from_index_token" in target_settings
+        and "iib_overwrite_from_index" not in target_settings
+    ) or (
+        "iib_overwrite_from_index_token" not in target_settings
+        and "iib_overwrite_from_index" in target_settings
+    ):
+        msg = (
+            "Either both or neither of 'iib_overwrite_from_index' and "
+            "'iib_overwrite_from_index_token' should be specified in target settings."
+        )
+        LOG.error(msg)
+        raise InvalidTargetSettings(msg)
+
+
+def task_iib_add_bundles(
+    bundles, archs, index_image, deprecation_list, signing_key, hub, task_id, target_settings
+):
+    """
+    Perform all the necessary actions for the 'PushAddIIBBundles' entrypoint.
+
+    Args:
+        bundles ([str]):
+            Bundles to add to the index image.
+        archs ([str]):
+            Architectures to build the index image for.
+        index_image (str):
+            Index image to add the bundles to.
+        deprecation_list ([str]):
+            Bundles to deprecate in the index image.
+        signing_key (str):
+            Signing key to be used.
+        hub (HubProxy):
+            Instance of XMLRPC pub-hub proxy.
+        task_id (str):
+            ID of the pub task.
+        target_settings (dict):
+            Dictionary containing settings necessary for performing the operation.
+    """
+    image_schema = "{host}/{namespace}/{repo}"
+    verify_target_settings(target_settings)
+
+    # Build new index image in IIB
+    build_details = OperatorPusher.iib_add_bundles(
+        bundles=bundles,
+        archs=archs,
+        index_image=index_image,
+        deprecation_list=deprecation_list,
+        target_settings=target_settings,
+    )
+
+    index_image_repo = image_schema.format(
+        host=target_settings.get("quay_host", "quay.io").rstrip("/"),
+        namespace=target_settings["quay_namespace"],
+        repo=get_internal_container_repo_name(target_settings["quay_operator_repository"]),
+    )
+    _, tag = build_details.index_image.split(":", 1)
+    dest_image = "{0}:{1}".format(index_image_repo, tag)
+
+    # Sign image
+    sig_handler = OperatorSignatureHandler(hub, task_id, target_settings)
+    sig_handler.sign_task_index_image(build_details, signing_key, build_details.index_image)
```

Comment:
I think build_details.index_image_resolved is the right one to be signed since build_details.index_image could be overwritten.

Reply:
From what I understand of the hello-operator guild discussion, new index image is always built as iib:<task_id>. So I guess I'll have to change the image name to "iib" to ensure that stable image is used. @midnightercz can you confirm or (if I'm wrong) explain the proper procedure?

Reply:
I changed the implementation to use intermediate index image